src/bun.js/node/fs_events.zig

```diff
@@ -107,6 +107,12 @@ var fsevents_mutex: Mutex = Mutex.init();
 var fsevents_default_loop_mutex: Mutex = Mutex.init();
 var fsevents_default_loop: ?*FSEventsLoop = null;
 
+fn symlinkPath(path: string, buffer: *[bun.MAX_PATH_BYTES + 1]u8) ?string {
```

Comment:
symlinks can have multiple levels

zig's std library here is not reliable due to usage of `catch unreachable`

Do we have a file descriptor for this or nah? if we do, we can use the code path for getting the path from the file

I think it's also worth looking through FSEvents to see if they expose a way to tell you this information in their events

Reply:
we have the fd because we open it to see if its a directory, FSEvents already resolve the path for me, I only need the dest path if its a symlink so I can check against the watcher.

Reply:
we want the resolved symlink path, right?  if yes, use the macOS function for getting the fully resolved symlink path  instead of readlink(). readlink() is only one hop. you could have symlink -> symlink -> dir and this would only catch the 2nd symlink.

Reply:
`std.os.getFdPath` did the trick also add a test for this in https://github.com/oven-sh/bun/pull/3846/commits/d2ed85ec96c43fd0be74e60c7dc998d37a0d1b0e